### PR TITLE
Fix rngs used for training

### PR DESCRIPTION
**Pull request by xibinliu**
# Description
Fix rngs used for training.
Also fix the unit tests to make them run in different tpu VMs

# Tests

Run a training successfully.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: tests/maxtext_utils_test.py [hunk 1]
```diff
@@ -405,7 +405,7 @@ def test_multi_axis_sharding_pass(self):
     multi-dimensional mesh passes the assertion.
     """
     # Create a mesh shape for a 5D mesh.
-    devices = np.array(jax.devices()).reshape((4, 1, 1, 1, 1))
+    devices = np.array(jax.devices()[:4]).reshape((4, 1, 1, 1, 1))
```
Comment (RissyRan) [inline tests/maxtext_utils_test.py]: I guess you met issue when testing on other tpu topology? Shall we instead do something like

```
num_devices = jax.device_count()
devices = np.array(jax.devices()).reshape((num_devices, 1, 1, 1, 1))
```

File: tests/data_loader_test.py [hunk 2]
```diff
@@ -43,7 +43,8 @@ def setUp(self):
         enable_rampup_batch_size=True,
         per_device_batch_size_start=1.0,
         per_device_batch_size_increment=1.0,
-        global_rampup_samples=60,
+        # global_rampup_samples: (rampup increment number) * (Samples for initial 5 steps)
+        global_rampup_samples=3 * (1 * jax.device_count() * 5),
```
Comment (RissyRan) [inline tests/data_loader_test.py:47]: @aireenmei could you help take a quick look?

File: tests/maxtext_utils_test.py (hunk 1 above)
Comment (xibinliu) [reply tests/maxtext_utils_test.py]: Good suggestion. I made the change and tests can pass on the other VM with 8 TPU devices.
